### PR TITLE
IPS stats/v3

### PR DESCRIPTION
**Pull request by victorjulien**
```json
{
  "accepted": 301661,
  "blocked": 219,
  "rejected": 0,
  "replaced": 0,
  "drop_reason": {
    "decode_error": 0,
    "defrag_error": 0,
    "defrag_memcap": 0,
    "flow_memcap": 0,
    "flow_drop": 143,
    "applayer_error": 0,
    "applayer_memcap": 0,
    "rules": 3,
    "threshold_detection_filter": 0,
    "stream_error": 68,
    "stream_memcap": 0,
    "stream_midstream": 5,
    "nfq_error": 0,
    "tunnel_packet_drop": 0
  }
}
```

https://redmine.openinfosecfoundation.org/issues/4756
https://redmine.openinfosecfoundation.org/issues/6230

replaces #9284, fixing counters updates

### CODE REVIEW

**Comment by suricata-qa** [comment]
Information: QA ran without warnings. 

Pipeline 15384

File: src/decode.h [hunk 1]
```diff
@@ -403,6 +403,7 @@ enum PacketDropReason {
     PKT_DROP_REASON_STREAM_MIDSTREAM,
     PKT_DROP_REASON_NFQ_ERROR,    /**< no nfq verdict, must be error */
     PKT_DROP_REASON_INNER_PACKET, /**< drop issued by inner (tunnel) packet */
+    PKT_DROP_REASON_MAX,
```
Comment (jufajardini) [inline src/decode.h:406]: If the idea is to help users figure out what's happening, wouldn't it make sense to have a drop reason specifically for stream reassembly memcap?

**Comment by jufajardini** [comment]
Would be good to have an SV test showcasing these new stats counters. I could take that up.

File: src/decode.h (hunk 1 above)
Comment (victorjulien) [reply src/decode.h:406]: yeah agree, but that is outside of the scope of this PR. Can you do a ticket for that?

Comment (jufajardini) [reply src/decode.h:406]: ^^ https://redmine.openinfosecfoundation.org/issues/6235

**Comment by victorjulien** [comment]
Merged in #9288

**Pull request closed without merging**